packages/x-date-pickers-pro/src/internals/utils/validation/validateDateTimeRange.ts

```diff
@@ -24,7 +24,18 @@ export const validateDateTimeRange: Validator<
 > = ({ props, value, adapter }) => {
   const [start, end] = value;
 
-  const { shouldDisableDate, ...otherProps } = props;
+  const { shouldDisableDate, disableNonContiguousRanges, ...otherProps } = props;
+
+  if (!!start && !!end && disableNonContiguousRanges && shouldDisableDate) {
+    let current = start;
+
+    while (adapter.utils.isBefore(current, adapter.utils.addDays(end, 1))) {
+      if (shouldDisableDate(current, 'start')) {
+        return ['nonContiguousRanges', 'nonContiguousRanges'];
+      }
+      current = adapter.utils.addDays(current, 1);
+    }
+  }
```

Comment:
This validation is a copy-paste of the one in `validateDateRange.ts`, because in the future (or perhaps in this PR) we might want to validate contiguous time ranges and additional validations would be necessary.